com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariableBase.cs

```diff
@@ -51,6 +51,7 @@ public void Initialize(NetworkBehaviour networkBehaviour)
         public virtual void SetDirty(bool isDirty)
         {
             m_IsDirty = isDirty;
+            m_NetworkBehaviour.NetworkManager.MarkNetworkObjectDirty(m_NetworkBehaviour.NetworkObject);
```

Comment:
In testing on the project I'm working on, it's possible for this to be null, which appears to happen if this is set before `Initialize()`. Probably worth checking dirty in `Initialize()` too.

There aren't any use cases with `false` now, but it might also be good to check `isDirty`.